OasisDEX/oasis-earn-sc
# :bug: Flashloans adjustments for Aave Like Protocols

## Ticket URL

[Shortcut](https://app.shortcut.com/oazo-apps/story/11917/frontend-library-issues-with-flashloans)

## Description of Changes
- Return flash loan information with `Multiply Strategy`
- Possibility to force flash loan token for Aave & Spark (Although that will work only on L2s)
- Refactor the Close operation. After depositing a flash loan, we were taking the MAX_UNIT amount of collateral. In the case of short positions (for example, with DAI on Mainnet), that would break the transaction. I have changed the behavior, and now, when the collateral and flash loan assets are the same, I withdraw only the necessary amount for the swap. 

## How to Test

Please try to play around with Aave v3 and Spark on the [preview deployment](https://kh8439-3000.csb.app/). In the order information you will see the flashloan amount. 

## PR Definition of Done

Please ensure the following requirements have been met before marking the PR as ready for review:

- [x] All checks are passing
- [x] PR is linked to a corresponding ticket
- [x] PR title is clear and concise
- [x] Code has been self-reviewed and any fixes or improvements noted (See Code review standards in Notion)
